mongoengine/queryset/base.py

```diff
@@ -566,6 +568,8 @@ def update(
                 for u in update["__raw__"]
             ]
         else:
+            if "array_filters" in update:
```

Comment:
array_filters is provided in the signature on line 534 so this will never be reached and should be removed